### PR TITLE
encoding.utf8: Add is_number

### PR DESCRIPTION
**Pull request by ChAoSUnItY**
- Add `is_number` function for testing if rune is either unicode number or in category N.
- Add test

```v
fn test_is_number() {
	for ra in `a` .. `z` {
		assert utf8.is_number(ra) == false
	}

	for ra in `A` .. `Z` {
		assert utf8.is_number(ra) == false
	}

	for ra in `0` .. `1` {
		assert utf8.is_number(ra) == true
	}

	assert utf8.is_number(`\u2164`) == true
	assert utf8.is_number(`\u2188`) == true
	assert utf8.is_number(`\u3029`) == true
}
```

### CODE REVIEW

File: vlib/encoding/utf8/utf8_tables.v [hunk 1]
```diff
@@ -1146,6 +1146,145 @@ const white_space_table = RangeTable{
 	latin_offset: 2
 }
 
+const number_table = RangeTable{
+	r16: [
+		Range16{0x0030, 0x0039, 1},
```
Comment (Delta456) [inline vlib/encoding/utf8/utf8_tables.v:1151]: Can you add comments for all these number tables for reference? I think it would be really handy.
